src/shell/mod.rs

```diff
@@ -322,6 +326,12 @@ async fn resolve_redirect_pipe(
     stderr.clone(),
   )
   .await;
+  let words = match words {
+    Ok(word) => word,
+    Err(err) => {
+      return Err(err.into_exit_code(stderr));
+    }
+  };
```

Comment:
Nitpick: seems like `let/else` could work nicely in several places in this file.

Reply:
do you know how to use let/else and get the error?

Reply:
Ah, good point. Disregard.